src/store.rs

```diff
@@ -286,6 +288,18 @@ pub struct ScalarPtr<F: LurkField>(F, F);
 
 impl<F: LurkField> Copy for ScalarPtr<F> {}
 
+#[cfg(not(target_arch = "wasm32"))]
+impl<Fr: LurkField> Arbitrary for ScalarPtr<Fr> {
+    type Parameters = ();
+    type Strategy = BoxedStrategy<Self>;
+
+    fn arbitrary_with(_args: Self::Parameters) -> Self::Strategy {
+        any::<(ExprTag, FWrap<Fr>)>()
+            .prop_map(|(tag, val)| ScalarPtr::from_parts(Fr::from(tag as u64), val.0))
```

Comment:
Would it make more sense to only create well-formed `ScalarPtr`s — that is, those with valid tags? I guess for testing purposes, it's valuable to have malformed also, but perhaps at a much lower frequency?

Do let me know if I'm misunderstanding or making bad assumptions about how this should work.

Reply:
For clarity: I do think having a way to generate bad or potentially bad values is generally useful, especially when testing circuits.

Reply:
> Would it make more sense to only create well-formed ScalarPtrs — that is, those with valid tags? I guess for testing purposes, it's valuable to have malformed also, but perhaps at a much lower frequency?

Yes, naturally.

`Arbitrary` is only one distinguished strategy for creating values of type T. The number of strategies for this type is unbounded, and does not have to involve the `Arbitrary` trait.

In other terms, I can have:
```
struct T {}
impl Arbitrary for T { ... }
fn my_strategy_for_T() -> impl Strategy<Value = T> { ... }
```
and at point of use:
```
proptest! {
  #[test]
  fn my_test_with_arbitrary_T(x in any::<T>()) { // uses Arbitrary
      ... 
  }
  
  #[test]
  fn my_test_with_custom_strategy(x in my_strategy_for_T()) { // does not use Arbitrary
      ....
  }
}
```

Reply:
Yes, this makes perfect sense and is along the lines of what I ended up thinking we would do. Thank you for the explanation.